### ISSUE TITLE
Cop idea: Accessing Elements Directly

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12350]
## Description

When accessing elements of a collection, avoid direct access via `[n]` by using an alternate form of the reader method if it is supplied. This guards you from calling `[]` on `nil`.

https://rubystyle.guide/#accessing-elements-directly

## Describe the solution you'd like

```ruby
/caaaa(.)t/ =~ 'cat'

# bad
Regexp.last_match[1]

# good
Regexp.last_match(1)
```

**Comment by koic** [comment]
Unfortunately, there is currently no way to determine if a call to `collection[0]` can be replaced with `collection(0)` using RuboCop's static analysis.
